src/views/Quickstart/index.jsx

```diff
@@ -0,0 +1,469 @@
+      },
+    },
+  ],
+};
+const baseCmd = [
+  'git clone {{event.head.repo.url}} repo',
+  'cd repo',
+  'git config advice.detachedHead false',
+  'git checkout {{event.head.sha}}',
+];
+const cmdDirectory = (type, org = '<YOUR_ORG>', repo = '<YOUR_REPO>') =>
+  ({
+    node: [
+      '/bin/bash',
+      '--login',
+      '-c',
+      baseCmd.concat(['npm install .', 'npm test']).join(' && '),
+    ],
+    python: [
+      '/bin/bash',
+      '--login',
+      '-c',
+      baseCmd.concat(['pip install tox', 'tox']).join(' && '),
+    ],
+    'rust:latest': [
+      '/bin/bash',
+      '-c',
+      baseCmd.concat(['rustc --test unit_test.rs', './unit_test']).join(' && '),
+    ],
+    golang: [
+      '/bin/bash',
+      '--login',
+      '-c',
+      [
+        `mkdir -p /go/src/github.com/${org}/${repo}`,
+        `cd /go/src/github.com/${org}/${repo}`,
+        'git init',
+        'git fetch {{ event.head.repo.url }} {{ event.head.ref }}',
+        'git config advice.detachedHead false',
+        'git checkout {{ event.head.sha }}',
+        'go install',
+        'go test ./...',
+      ].join(' && '),
+    ],
+  }[type]);
+
+@hot(module)
+@withApollo
+@withStyles(theme => ({
+  orgRepo: {
+    display: 'flex',
+    alignItems: 'center',
+    ...theme.mixins.gutters(),
+  },
+  separator: {
+    padding: theme.spacing.double,
+    paddingBottom: 0,
+  },
+  checkIcon: {
+    fill: theme.palette.success.main,
+  },
+  errorIcon: {
+    fill: theme.palette.error.main,
+  },
+  resetButton: {
+    ...theme.mixins.fab,
+  },
+  errorPanels: {
+    marginTop: theme.spacing.double,
+  },
+  iconContainer: {
+    marginLeft: theme.spacing.unit,
+  },
+}))
+export default class QuickStart extends Component {
+  static initialState = {
+    events: new Set([
+      'pull_request.opened',
+      'pull_request.reopened',
+      'pull_request.synchronize',
+    ]),
+    taskName: '',
+    taskDescription: '',
+  };
+
+  state = {
+    ...QuickStart.initialState,
+    owner: '',
+    repo: '',
+    access: 'collaborators',
+    image: 'node',
+    commands: cmdDirectory('node'),
+    commandSelection: 'standard',
+    installedState: null,
+  };
+
+  handleAccessChange = ({ target: { value } }) => {
+    this.setState({ access: value, editorValue: null });
+  };
+
+  handleEventsSelection = ({ target: { value } }) => {
+    const events = new Set(this.state.events);
+
+    events.has(value) ? events.delete(value) : events.add(value);
+
+    this.setState({
+      events,
+      editorValue: null,
+    });
+  };
+
+  handleInputChange = ({ target: { name, value } }) => {
+    this.setState({ [name]: value, editorValue: null });
+  };
+
+  handleCommandsChange = ({ target: { value } }) => {
+    this.setState({
+      commandSelection: value,
+      commands: value === 'standard' ? cmdDirectory(this.state.image) : [],
+      editorValue: null,
+    });
+  };
+
+  getInstalledState = debounce(async (owner, repo) => {
```

Comment:
I think this is the ideal situation where a `<Query>` component makes sense. We should be able to use the state to drive the variables passed to the Query and use a render prop to drive the rendering based on that. Make sense?

Reply:
When the `Query` component is rendered it will automatically fire off the query which is not my intention to do. I went with the manual approach so that I could make sure the request uses a debouncer for the query call. For delaying queries, Apollo [suggests](https://www.apollographql.com/docs/react/essentials/queries.html#manual-query) using `client.query()`. How would you use the `Query` component while making sure the query is debounced?

Reply:
Wouldn't you just render the component or not based on the value of the props, i.e. if the props weren't there, don't render the Query component? If not, feel free to move forward with this approach, just let me know which way so I can review or not. :)

Reply:
Ah, `getInstalledState` is actually triggered everytime a user types something inside the repository or organization text field. It is being called by `handleOrgRepoChange`. I will move forward with this approach :)